client/driver/logging/rotator.go

```diff
@@ -88,12 +91,10 @@ func (f *FileRotator) Write(p []byte) (n int, err error) {
 		if remainingSize < int64(len(p[n:])) {
```

Comment:
This should scan backwards through the buffer for a `\n` and write/flush everything before that.  Whatever comes after that last `\n` should be moved to the beginning of the buffer and written out on the next write (as long as the line length is less than the buffer length, in which case you need to truncate).  This only needs to happen on file rotation, not on write.  I'll open an issue re: this in a sec.